module/Kanbanize/src/Kanbanize/Service/Importer.php

```diff
@@ -0,0 +1,442 @@
+						$task->complete($this->requestedBy);
+					case Task::STATUS_COMPLETED:
+					case Task::STATUS_IDEA:
+						$task->execute($this->requestedBy);
+				}
+				break;
+			case strtoupper(Task::STATUS_COMPLETED):
+				switch ($task->getStatus()){
+					case Task::STATUS_CLOSED:
+						$task->accept($this->requestedBy);
+					case Task::STATUS_ACCEPTED:
+					case Task::STATUS_ONGOING:
+						$task->complete($this->requestedBy);
+						break;
+					case Task::STATUS_IDEA:
+						$task->execute($this->requestedBy);
+						$task->complete($this->requestedBy);
+				}
+				break;
+			case strtoupper(Task::STATUS_ACCEPTED):
+				switch ($task->getStatus()){
+					case Task::STATUS_IDEA:
+						$task->execute($this->requestedBy);
+					case Task::STATUS_ONGOING:
+						$task->complete($this->requestedBy);
+					case Task::STATUS_COMPLETED:
+					case Task::STATUS_CLOSED:
+						$task->accept($this->requestedBy);
+				}
+				break;
+			case strtoupper(Task::STATUS_CLOSED):
+				switch ($task->getStatus()){
+					case Task::STATUS_IDEA:
+						$task->execute($this->requestedBy);
+					case Task::STATUS_ONGOING:
+						$task->complete($this->requestedBy);
+					case Task::STATUS_COMPLETED:
+						$task->accept($this->requestedBy);
+					case Task::STATUS_ACCEPTED:
+						$task->close($this->requestedBy);
+				}
+				break;
+		}
+		$task->setColumnName($columnName, $this->requestedBy);
+	}
+	/**
+	 * 
+	 * @param Task $task
+	 * @param string $username
+	 */
+	private function updateTaskOwner(Task $task, $username){
+		$users = $this->userService->findUsers(['kanbanizeusername' => $username]);
+		$new_owner = array_shift($users);
+		if (is_null($new_owner)){
+			$task->removeOwner($this->requestedBy);
+		}elseif (!is_null($task->getOwner())){
+			if(!$task->hasMember($new_owner)){
+				$task->addMember($new_owner, TaskMember::ROLE_MEMBER, $this->requestedBy);
+			}
+			$task->changeOwner($new_owner, $this->requestedBy);
+		}elseif ($this->ownershipToBeAdded($task, $new_owner)){
+			$task->addMember($new_owner, TaskMember::ROLE_OWNER, $this->requestedBy);
+		}elseif($this->ownershipToBeAssigned($task, $new_owner)){
+			$task->changeOwner($new_owner, $this->requestedBy);
+		}
+		$task->setAssignee($username, $this->requestedBy);
+	}
+	/**
+	 * @param Task $task
+	 * @param User $new_owner
+	 * @return boolean
+	 */
+	private function ownershipToBeAdded(Task $task, User $new_owner){
+		return is_null($task->getOwner()) && !$task->hasMember($new_owner);
+	}
+	/**
+	 * @param Task $task
+	 * @param User $new_owner
+	 * @return boolean
+	 */
+	private function ownershipToBeAssigned(Task $task, User $new_owner){
+		return is_null($task->getOwner()) && $task->hasMember($new_owner);
+	}
+	private function createFormattedSubject($projectName, $boardName){
+		return $projectName."/".$boardName;
+	}
+
+	private function deleteTasks(Stream $stream, $tasksFound){
+		$organization = $this->organizationService->findOrganization($this->organization->getId());
```

Comment:
In realtà mi servono entrambe: l'organizzazione del write model mi serve quando devo chiamare eventualmente una `KanbanizeStream::create`, quella del read model invece mi serve in questo caso, visto che la findTasks riceve come primo parametro `People\Entity\Organization` : si può pensare di togliere questa "tipizzazione" ?